### PR TITLE
Fix duplicating paths during remapping

### PR DESCRIPTION
**Pull request by BruRdgz**
Adds tests for #1752's bugs, along with a patch. Closes the issue.

### CODE REVIEW

File: tests/test_api.py [hunk 1]
```diff
@@ -526,6 +514,51 @@ def get_combined_filenames() -> set[str]:
...
+        self.start_import_stop(cov_bar, "bar")
+        cov_bar.save()
+
+        # Combine the data files in the opposite order.
+        cov2 = coverage.Coverage()
+        cov2.combine(data_paths=[".coverage.bar_cov.data", ".coverage.foo_cov.data"])
+        files2 = cov2.get_data().measured_files()
+
+        # The order of combining should not affect the resulting files.
+        assert set(files1) == set(files2)
```
Comment (BruRdgz) [inline tests/test_api.py:559]: In hindsight, this is pretty obvious; so that's a stupid check to take away.

File: tests/test_api.py [hunk 2]
```diff
@@ -500,18 +500,6 @@ def get_combined_filenames() -> set[str]:
             filenames = {relative_filename(f).replace("\\", "/") for f in data.measured_files()}
             return filenames
 
-        # Case 1: get the order right.
-        make_files()
-        self.make_file(".coveragerc", """\
-            [paths]
-            plugins =
-                plugins/
-                ci/girder/plugins/
```
Comment (BruRdgz) [inline tests/test_api.py:509]: This is weird; I didn't touch this, and these files remain on my local repository. Go figure, I'll check it out later.

**Review by BruRdgz — COMMENTED** [review]
Have to check those out.

File: coverage/files.py [hunk 3]
```diff
@@ -489,6 +489,9 @@ def map(self, path: str, exists:Callable[[str], bool] = source_exists) -> str:
 
         # If we get here, no pattern matched.
 
+        if self.relative:
```
Comment (nedbat) [inline coverage/files.py:492]: A key indication of whether the test is doing what it should:  comment out "the fix" and see if the test fails.  When I comment out these two lines, all the tests still pass.

File: tests/test_api.py [hunk 4]
```diff
@@ -500,18 +500,6 @@ def get_combined_filenames() -> set[str]:
             filenames = {relative_filename(f).replace("\\", "/") for f in data.measured_files()}
             return filenames
 
-        # Case 1: get the order right.
```
Comment (nedbat) [inline tests/test_api.py:503]: I don't understand why you removed this test code.

**Comment by nedbat** [comment]
Thanks, I fixed it like this: 1b19799edfbfc65dea254e8fcaa0dea23e9709f9

**Pull request closed without merging**